src/__mocks__/openWebSocket.js

```diff
@@ -0,0 +1,63 @@
+const openWebSocketModule = jest.genMockFromModule('../openWebSocket');
+
+const SUCCESS_URL = 'ws://localhost';
+const FAILING_URL = 'ws://localhost/oops';
+
+function openWebSocketMock(url) {
+	const _listeners = [];
+
+	const socket = {
```

Comment:
OK so here we're creating an mock WebSocket Object which is going to provide the same functions as the WebSocket which we'd get from 	'return new window.WebSocket(url);' in the real implementation. To enable us to do this you've had to split out the websocket creation into it's own file openWebSocket.js, so it can be overridden under __mocks__. This mock web socket implementation has to be able to inject a mock EXECUTE_STEP response at the right moment when simulateServerMessage is called from our unit tests.